lib/services/local/externalcloudaudit_watcher.go

```diff
@@ -0,0 +1,188 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package local
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+	"github.com/sirupsen/logrus"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/utils/retryutils"
+	"github.com/gravitational/teleport/lib/backend"
+	"github.com/gravitational/teleport/lib/defaults"
+)
+
+var ErrClusterExternalCloudAuditWatcherClosed = errors.New("cluster external cloud audit watcher closed")
+
+// ClusterExternalCloudAuditWatcherConfig contains configuration options for a ClusterExternalAuditWatcher.
+type ClusterExternalCloudAuditWatcherConfig struct {
+	// Backend is the storage backend used to create watchers.
+	Backend backend.Backend
+	// Log is a logger.
+	Log logrus.FieldLogger
+	// Clock is used to control time.
+	Clock clockwork.Clock
+	// OnChange is the action to take when the cluster ExternalCloudAudit
+	// changes.
+	OnChange func()
+}
+
+// CheckAndSetDefaults checks parameters and sets default values.
+func (cfg *ClusterExternalCloudAuditWatcherConfig) CheckAndSetDefaults() error {
+	if cfg.Backend == nil {
+		return trace.BadParameter("missing parameter Backend")
+	}
+	if cfg.Log == nil {
+		cfg.Log = logrus.StandardLogger().WithField(trace.Component, "ExternalCloudAudit.watcher")
+	}
+	if cfg.Clock == nil {
+		cfg.Clock = cfg.Backend.Clock()
+	}
+	if cfg.OnChange == nil {
+		return trace.BadParameter("missing parameter OnChange")
+	}
+	return nil
+}
+
+// ClusterExternalAuditWatcher is a light weight backend watcher for the cluster external audit resource.
+type ClusterExternalAuditWatcher struct {
+	ClusterExternalCloudAuditWatcherConfig
+	retry retryutils.Retry
+	sync.Mutex
```

Comment:
Similarly, should other packages be able to lock/unlock the watcher's mutex, or is the mutex an implementation detail that should not be exposed to the outside world?